src/test/scala/com/amazon/deequ/comparison/ReferentialIntegrityTest.scala

```diff
@@ -474,4 +558,123 @@ class ReferentialIntegrityTest extends AnyWordSpec with SparkContextSpec {
       assert(result.isInstanceOf[ComparisonSucceeded])
     }
   }
+
+  "Referential Integrity Row Level Test" should {
+    "works for multiple columns" in withSparkSession { spark =>
+      import spark.implicits._
+
+      val rdd1 = spark.sparkContext.parallelize(Seq(
+        (1, "New York", "NY"),
+        (2, "Wisconsin", "WI"),
+        (3, "Texas", "TX"),
+        (4, "Canada", "CA"))) // Incorrect row
+      val ds1 = rdd1.toDF("id", "state name", "state")
+
+      val rdd2 = spark.sparkContext.parallelize(Seq(
+        (1, "New York", "NY"),
+        (2, "Wisconsin", "WI"),
+        (3, "Texas", "TX"),
+        (4, "California", "CA"))) // Reference has correct row
+      val ds2 = rdd2.toDF("id", "state name", "state")
+
+      val cols = Seq("state name", "state")
+      val outcomeCol = "row_level_outcome"
+
+      val result = ReferentialIntegrity.subsetCheckRowLevel(ds1, cols, ds2, cols, Some(outcomeCol))
+      assert(result.isRight)
+      val outcomes = result.right.get.select(outcomeCol).collect().toSeq.map { r => r.get(0) }
+      assert(outcomes == Seq(true, true, false, true))
```

Comment:
maybe also assert that if I drop `"row_level_outcome"` I get a dataframe with exactly what's in `rdd1`?

Reply:
Done.